### PR TITLE
Remove global config (garage.config)

### PR DESCRIPTION
**Pull request by zequnyu**
- removed global configuration (garage.config) 
- deleted unused code in experiment.py

#382.

### CODE REVIEW

File: examples/tf/dqn_cartpole.py [hunk 1]
```diff
@@ -22,17 +22,23 @@ def run_task(*_):
         n_epoch_cycles = 10
         sampler_batch_size = 500
         num_timesteps = n_epochs * n_epoch_cycles * sampler_batch_size
+
```
Comment (ryanjulian) [inline examples/tf/dqn_cartpole.py]: all of these spacing changes don't seem relevant to this PR. please remove them.

File: examples/tf/her_ddpg_fetchreach.py [hunk 2]
```diff
@@ -71,7 +71,7 @@ def run_task(*_):
 
         runner.setup(algo=ddpg, env=env)
 
-        runner.train(n_epochs=50, n_epoch_cycles=20)
+        runner.train(n_epochs=50, n_epoch_cycles=20, batch_size=100)
```
Comment (ryanjulian) [inline examples/tf/her_ddpg_fetchreach.py]: this doesn't seem relevant to the PR title. please remove.

File: examples/tf/trpo_cartpole_batch_sampler.py [hunk 3]
```diff
@@ -46,8 +46,4 @@ def run_task(*_):
         runner.train(n_epochs=100, batch_size=4000, plot=False)
 
 
-run_experiment(
-    run_task,
-    snapshot_mode='last',
-    seed=1,
-)
+run_experiment(run_task, snapshot_mode='last', seed=1)
```
Comment (ryanjulian) [inline examples/tf/trpo_cartpole_batch_sampler.py]: this change is not relevant to the PR. please remove.

File: src/garage/experiment/__init__.py [hunk 4]
```diff
@@ -3,15 +3,11 @@
 from garage.experiment.experiment import to_local_command
 from garage.experiment.experiment import variant
 from garage.experiment.experiment import VariantGenerator
-from garage.experiment.snapshotter import Snapshotter
-
-# LocalRunner needs snapshotter to be imported, so we have to use a strange
-# import order here
-snapshotter = Snapshotter()
```
Comment (ryanjulian) [inline src/garage/experiment/__init__.py]: this change is not related to the PR. did you introduce a merge commit, or branch from one of your other PRs?

File: src/garage/experiment/experiment.py [hunk 5]
```diff
@@ -223,64 +216,37 @@ def run_experiment(method_call=None,
...
-    :param terminate_machine: Whether to terminate machine after experiment
-     finishes. Only used when using mode="ec2". This is useful when one wants
-     to debug after an experiment finishes abnormally.
-    :param periodic_sync: Whether to synchronize certain experiment files
-     periodically during execution.
-    :param periodic_sync_interval: Time interval between each periodic sync,
-     in seconds.
+
+    Args:
+        method_call (object): A method call.
```
Comment (ryanjulian) [inline src/garage/experiment/experiment.py]: are `method_call` and `batch_tasks` both callables?

File: src/garage/experiment/experiment.py [hunk 6]
```diff
@@ -223,64 +216,37 @@ def run_experiment(method_call=None,
...
+        use_tf (bool): Used along with the Theano and GPU configuration
+            when using TensorFlow
+        use_gpu (bool): Whether the launched task is running on GPU.
+            This triggers a few configuration changes including certain
+            environment flags.
+        use_cloudpickle (bool): Whether to use cloudpickle or not.
+        pre_commands (str): Pre commands to run the experiment.
+
     """
     assert method_call is not None or batch_tasks is not None, \
```
Comment (ryanjulian) [inline src/garage/experiment/experiment.py]: please change this to split the lines by enclosing in () (per PEP8)

File: src/garage/experiment/experiment.py [hunk 7]
```diff
@@ -328,14 +289,17 @@ def run_experiment(method_call=None,
...
                                                exp_count)
         if task.get('log_dir', None) is None:
-            task['log_dir'] = (
-                config.GARAGE_LOG_DIR + '/local/' + exp_prefix.replace(
-                    '_', '-') + '/' + task['exp_name'])
+            garage_project_path = osp.abspath(
+                osp.join(osp.dirname(__file__), '..'))
+            garage_log_dir = osp.join(garage_project_path, 'data')
+
+            task['log_dir'] = (garage_log_dir + '/local/' + exp_prefix.replace(
```
Comment (ryanjulian) [inline src/garage/experiment/experiment.py]: please use `.format()` for string interpolation.

File: src/garage/experiment/experiment.py [hunk 8]
```diff
@@ -344,45 +308,15 @@ def run_experiment(method_call=None,
...
-            script=script,
-            python_command=python_command,
-            sync_s3_pkl=sync_s3_pkl,
-            sync_log_on_termination=sync_log_on_termination,
-            periodic_sync=periodic_sync,
-            periodic_sync_interval=periodic_sync_interval,
-            sync_all_data_node_to_s3=sync_all_data_node_to_s3,
-        )
-    elif mode == 'local':
+    if mode == 'local':
```
Comment (ryanjulian) [inline src/garage/experiment/experiment.py]: if seems that you can remove `mode` entirely?

File: src/garage/experiment/experiment.py [hunk 9]
```diff
@@ -328,14 +289,17 @@ def run_experiment(method_call=None,
...
-        params = dict(kwargs)
+
         if task.get('exp_name', None) is None:
             task['exp_name'] = '%s_%s_%04d' % (exp_prefix, timestamp,
                                                exp_count)
         if task.get('log_dir', None) is None:
-            task['log_dir'] = (
-                config.GARAGE_LOG_DIR + '/local/' + exp_prefix.replace(
-                    '_', '-') + '/' + task['exp_name'])
+            garage_project_path = osp.abspath(
```
Comment (ryanjulian) [inline src/garage/experiment/experiment.py]: this is not a great way to do this. it is doing to put data inside a git repository (or possibly installation directory, if garage is installed using pip). 

I think the default should be `'data'`, in the same directory as the launcher script.

**Comment by zequnyu** [comment]
I think I created this PR based on the `snapshotter` one. I'll merge that one before I continue the work for this PR.

**Comment by zequnyu** [comment]
All the irrelevant changes were from the `snapshotter` commit. I'll merge it before this. Please review this later.

File: src/garage/experiment/experiment.py (hunk 5 above)
Comment (zequnyu) [reply src/garage/experiment/experiment.py]: `batch_tasks` is a dict containing `method_call` and other info.
```python3
batch_tasks = [
            dict(
                kwargs,
                pre_commands=pre_commands,
                method_call=method_call,
                exp_name=exp_name,
                log_dir=log_dir,
                env=env,
                variant=variant,
                use_cloudpickle=use_cloudpickle)
        ]
```

Comment (ryanjulian) [reply src/garage/experiment/experiment.py]: what is the type of method_call?

**Comment by ryanjulian** [comment]
I will have to force-push this past the CI.

Please add a test verifying that the logs go into the correct directory by default.

File: src/garage/experiment/experiment.py (hunk 5 above)
Comment (zequnyu) [reply src/garage/experiment/experiment.py]: Emmm it's a function. Shall I say `func` or `callable`?

**Comment by zequnyu** [comment]
This is ready for force-push.

File: src/garage/experiment/experiment.py (hunk 5 above)
Comment (ryanjulian) [reply src/garage/experiment/experiment.py]: callable
